### ISSUE TITLE
Exchange Listing

### ISSUE THREAD

**Issue by cryptokat** [semuxproject/semux#621]
I am opening up this discussion since we are currently flooded by this question everywhere: `When will be the exchange listing of Semux?`

From a technical perspective, we need to satisfy the following requirements before we can make a successful exchange listing:

- **Multisignature Wallet**
A professional exchange may like to keep customers' funds inside of a multisignature cold wallet, in order to avoid [incidents](https://www.coindesk.com/tag/hacks/) we've seen in the past few years.
see: #617 #618 
- **DDoS Resistance**
Semux is utilizing the PBFT algorithm where `f = 33` to reach consensus among validators, which means one botnet operator / malicious validator may try to profit from taking down 33 or more validators by making more block rewards or threatening the foundation. 
see: #98 #590  
- ~~~**Ability To Manage a Large Number of Accounts**~~~
~~~A popular exchange may need to manage millions of accounts / private keys of Semux for their customers.~~~
see: #534

The latest todo list is tracked in: https://github.com/semuxproject/semux/projects/3

Please feel free to leave your comments below.

**Comment by orogvany** [comment]
I agree with much of what's said.  I've commented my thoughts on most of those, but for #534 in particular, it seems rather moot.

A popular exchange should *not* keep millions of private keys active in one wallet, as a security risk.  they would only need to generate millions of keys offline, and check their balances when deposits are made.  One can check amounts without having them in wallet.  Keeping them all hot would be terribly bad practice.  
Bloom filters or similar could be used to filter new transactions to monitor deposits.  This could be supported internally if we wish, but feels rather central to just exchanges.

**Comment by cryptokat** [comment]
@orogvany Yes, I agree with you one exchange might have their own way to maintain private keys. A more scalable way would be keeping those keys in their own infrastructure, then sign a transaction offline when they need to move customer's funds.

**Comment by orogvany** [comment]
Multisig seems like a no brainer, we would surely benefit from that.

I'm afraid I don't believe in #590.  The argument is that adding some energy wasting algorithm will magically prove that the server is secure against attack is a fallacy.  A large machine does not imply any security.  I could argue in the same vein that we should ban all windows servers, but it wouldn't improve the security of the network.  Frankly, any machine has vulnerabilities to attack.  Working on hardening consensus to be more fault tolerant is a better use of our time and resources.  

#98 is a good feature, particularly for long term as whale validators lose interest, or don't notice server is down (see testnet :D).  Figuring out all the edgecases to avoid abuse will take time.

**Comment by cryptokat** [comment]
I created a Kanban to track TODOs for exchange listing. Please let me know if you feel like there are more tasks to be added.

https://github.com/semuxproject/semux/projects/3

**Comment by AlexAltertrader** [comment]
https://github.com/semuxproject/semux/issues/611 ?
https://github.com/semuxproject/semux/issues/590 ?

**Comment by cryptokat** [comment]
Hello @AlexAltertrader 

I agree with @orogvany that we will be able to maintain the availability of Semux consensus by fixing #98. 

#611 and #590 would be further hardening to the consensus which might not be necessarily required for exchange listing.

**Comment by AlexAltertrader** [comment]
#590 is not about security issues. And that now the validator of SEM can be run on any shit, falling from a slight breeze blow.
Either SEM remains a children's sandbox falling right now from several thousand transactions sent at once (if you do this now, the blockchain SEM will die and without the hardfork it will not run), or SEM will become something big.

**Comment by orogvany** [comment]
I would suggest #624 to list of possible improvements for overall network reliability and failure protection.  It can also come later.

**Comment by semuxdev** [comment]
I think many of the points are valuable. It's not only for exchange listing, but also to make semux better.

**Comment by spaceblockstoken** [comment]
Another issue we need to address is exchange hacked situation.  How we protect legitimate owners?

**Comment by orogvany** [comment]
@spaceblockstoken i dont know what we could do to enforce an exchanges security apart from just supporting multisig.  If exchange gets hacked, theres no risk to holders outside of exchange today.  As with any crypto, leaving money on exchange is poor security wise as you are entrusting that exchange with your funds and its care

**Comment by semuxdev** [comment]
Optionally, we can use multi-sig wallets once the EVM is integrated.

**Comment by cryptokat** [comment]
https://stocks.exchange/trade/SEM/BTC
